doc/users-guide/base/configs.rst

```diff
@@ -0,0 +1,220 @@
+
+Setting ``config`` defaults on the compiler's command-line also
+supports more radical alterations to the code.  For example, we
+could change *age* from an integral type to a floating point type:
+
+.. literalinclude:: examples/users-guide/base/configs2.compopts-fake
+  :language: text
+  :lines: 4
+
+Note that ``config`` declarations are processed in the order that they
+appear in the source code and that any initializing expressions must
+make sense when considered in that original context.  Thus, the
+command-line overrides for a given ``config`` can refer to other
+declarations that precede that declaration in the source code.  For
+instance, we can set *age* in terms of the ``config param`` *size*
+since it appeared earlier in the code:
+
+.. literalinclude:: examples/users-guide/base/configs2.compopts-fake
+  :language: text
+  :lines: 7
+
+However, the reverse would not be possible since *age* had not been
+established when *size* is declared.
+
+Similarly, subsequent declarations in the source code can themselves
+be based on ``config`` declarations.  For instance, here are some
+declarations that are based on the ``config param`` and ``config
+type`` statements shown above and could follow them:
+
+.. literalinclude:: examples/users-guide/base/configs2.chpl
+  :language: chapel
+  :lines: 9-11
+
+
+Compile time overrides for ``config var`` and ``config const``
+--------------------------------------------------------------
+
+Users can also specify default expressions for ``config var`` and
+``config const`` declarations on the compiler's command-line.  As with
+other compile-time overrides, complex or symbolic expressions can be
+used since the compiler is involved.  As an example, let's look at
+a variation on one of our earlier declarations:
+
+.. literalinclude:: examples/users-guide/base/configs3.chpl
+  :language: chapel
+  :lines: 1-2
+
+The followng compiler command can be used to override the source
+code initializer for *epsilon* so that it will be a function of
+*n* by default:
+
+.. literalinclude:: examples/users-guide/base/configs3.compopts
+  :language: text
+
+Such an initializer could not be specified on the executable
+command-line since the compiler is out of the picture by then.  At
+that point, the values for a ``config var`` or ``config const`` are
+restricted to the set of strings that can legally be cast to that
+type.
+
+Having compiled *epsilon* to be set in this way, if we specify
+a value for *n* on the executable's command line:
+
+.. literalinclude:: examples/users-guide/base/configs3.execopts-fake
+  :language: text
+  :lines: 2
+
+then *epsilon* will be computed in terms of it, resulting in the value:
+
+.. literalinclude:: examples/users-guide/base/configs3-n1000.good
+  :language: text
+  :lines: 2
+
+Note that the compiler's command-line override can still be
+re-overridden on the executable's command-line as usual:
+
+.. literalinclude:: examples/users-guide/base/configs2.compopts-fake
```

Comment:
The preceding text suggests we're going to see an example of overriding config vars on the executable's command line  But it includes only a line from a compopts file, showing an invocation of `chpl`.